cloudbuild/python-scripts/parse_logs.py

```diff
@@ -0,0 +1,501 @@
+    from a "filter" query contains only the desired (filtered) values.
+
+    Args:
+      gcs_log_object: GCS log object (name of the GCS Object containing the logs).
+      logs_as_string: Yarn application logs downloaded as a string.
+
+    Returns:
+      True if query result is seen in the file and does not violate filter
+      condition.
+      False in case no query result is obtained in the file (file as a string).
+
+    Raises:
+      RuntimeError: When filter condition is not met.
+    """
+
+    # Query records are of the format [ HOUR, DAY ].
+    # The pattern searches for records formatted the same way.
+    # A single space '\s' followed by a group (the HOUR), a ', ' and a space '\s'
+    # which is again followed by a group (the DAY)
+    query_records_pattern = r'\[\s(.*?),\s(.*?)\s\]'
+
+    # Find all matches of the pattern in the string.
+    matches = re.findall(query_records_pattern, logs_as_string)
+
+    # Extract and print all pairs of HOUR and DAY.
+    if matches:
+        print(f'[Log: parse_logs INFO] Query result obtained in {gcs_log_object}')
+        for match in matches:
+            hour = match[0].strip()
+            day = match[1].strip()
+
+            # Check if the records thus obtained follow the filter condition.
+            # Hardcoded check if HOUR and DAY are both = '17'.
+            if hour != '17' or day != '17':
+                raise RuntimeError(
+                    '[Log: parse_logs ERROR] Incorrect query result obtained!'
+                )
+    else:
+        # If no such matches are found.
+        print(
+            '[Log: parse_logs WARNING] No query result obtained in'
+            f' {gcs_log_object}'
+        )
+        return False
+    return True
+
+
+def get_blob_and_check_metric(
+    gcs_log_object,
+    cluster_temp_bucket,
+    metric_string,
+    end_of_metric_string,
+    query,
+):
+    """Method to extract the yarn logs file as a string and find metric string.
+
+    Args:
+      gcs_log_object: string
+      cluster_temp_bucket: name of the bucket (cluster temp bucket)
+      metric_string: The string to be found
+      end_of_metric_string: Delimiter at the end of the value.
+      query: Query string [if any] that was provided to the connector.
+
+    Returns:
+    Tuple [Metric Value (String), Is Query Result Present (Boolean)]
+      Metric value: which is the sum of metric values obtained from a file.
+       -1 in case metric is not seen in the file.
+      Is Query Result Present: Boolean value. True incase query result
+        is present in the file, False if not.
+    """
+    # Obtain the yarn logs as a string from the GCS bucket.
+    logs_as_string = ''
+    try:
+        storage_client = storage.Client()
+        bucket = storage_client.get_bucket(cluster_temp_bucket)
+        blob = bucket.blob(gcs_log_object)
+        logs_as_string = blob.download_as_text(encoding='latin-1')
+    except Exception as e:
+        print(
```

Comment:
Is there a logging framework we can use?

Reply:
Fixed.